go/proto/hydra/v1/deployment.proto

```diff
@@ -15,8 +14,11 @@ service DeploymentService {
 
 message DeployRequest {
   string deployment_id = 1;
-  string docker_image = 2;
-  optional string key_auth_id = 3;
+  optional string key_auth_id = 2;
+  // Build source fields, exactly one of (context_key, docker_image) must be set
+  optional string build_context_path = 3;
+  optional string dockerfile_path = 4;
+  optional string docker_image = 5;
```

Comment:
nit: can we use oneof here? 

since one of them is required either build_cointenxt_path + file path or docker image

Reply:
Yeah. I tried that a week ago and had a serialization issue with restate then I sticked with flat order like this. Maybe its fixed after new restate update.